### PR TITLE
bpo-37027: Return a proxy socket object from transp.get_extra_info('socket')

### PR DESCRIPTION
**Pull request by 1st1**
Return a safe to use proxy socket object from `transport.get_extra_info('socket')`

https://bugs.python.org/issue37027

### CODE REVIEW

**Review by asvetlov — APPROVED** [review]
asyncio saves at least "process" and "pipe" in extra info.
Should we wrap these things too?
Should we wrap sockets from `asyncio.Server()`?

**Comment by 1st1** [comment]
Andrew, please re-review this PR (and please hit the merge button if it's ready).  I've relaxed this PR - now all socket methods can be called as they are in 3.7, but we issue a warning that we won't support that in 3.9.

File: Lib/asyncio/trsock.py [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+
+    def dup(self):
+        return self._sock.dup()
+
+    def get_inheritable(self):
+        return self._sock.get_inheritable()
+
+    def shutdown(self, how):
+        # asyncio doesn't currently provide a high-level transport API
+        # to shutdown the connection.
```
Comment (asvetlov) [inline Lib/asyncio/trsock.py:75]: Technically there is `transport.write_eof()` for closing writing.
There is no API for closing reading though.

**Review by asvetlov — APPROVED** [review]

File: Lib/asyncio/trsock.py (hunk 1 above)
Comment (1st1) [reply Lib/asyncio/trsock.py:75]: Yes, and I know some people use `sock.shutdown` for that.  Should we add an API to transports?

Comment (asvetlov) [reply Lib/asyncio/trsock.py:75]: Maybe, to make the transition to new Python smoother.
Not urgent, though. Minor thing.

**Pull request merged**